src/evm.cuh

```diff
@@ -3406,8 +3412,119 @@ public:
 #ifdef TRACER
             tracer_t::print_tracer_data_t(arith, instances.tracers_data[idx], &instances.return_data[idx]);
 #endif
+        }
+
+
+        #ifdef COMPLEX_TRACER
+        touch_state_data_t prev_state, updated_state;
+        world_state_t world_state(arith, instances.world_state_data);
+        accessed_state_t accessed_state(&world_state);
+        prev_state.touch = new uint8_t[instances.world_state_data->no_accounts];
+        prev_state.touch_accounts.no_accounts = instances.world_state_data->no_accounts;
+        prev_state.touch_accounts.accounts = instances.world_state_data->accounts;
+
+
+        touch_state_t final_state(&prev_state, &accessed_state, arith);
+
+        updated_state.touch = new uint8_t[instances.touch_states_data->touch_accounts.no_accounts];
+        for (size_t idx = 0; idx < instances.touch_states_data->touch_accounts.no_accounts; idx++)
+        {
+            updated_state.touch[idx] = instances.touch_states_data->touch[idx];
+        }
+        updated_state.touch_accounts.no_accounts = instances.touch_states_data->touch_accounts.no_accounts;
+        updated_state.touch_accounts.accounts = instances.touch_states_data->touch_accounts.accounts;
+        touch_state_t state_from_updated(&updated_state, &accessed_state, arith);
+
+
+        final_state.update_with_child_state(state_from_updated);
+        print_touched_trie_accounts(arith, final_state._content);
+
+        delete[] prev_state.touch;
+        delete[] updated_state.touch;
+        prev_state.touch = nullptr;
+        updated_state.touch = nullptr;
+
+
+        #endif
+        if (cpu_world_state != nullptr){
+            delete cpu_world_state;
+            cpu_world_state = NULL;
+        }
+    }
+
+    __host__ static void print_touched_trie_accounts(arith_t &arith, touch_state_data_t *evm_instances)    {
```

Comment:
these test or debug only functions should be placed outside the current file which contains core EVM logic to reduce cluttering. Any suggestions?